src/Fetch/Server.php

```diff
@@ -135,6 +142,33 @@ public function __construct($serverPath, $port = 143, $service = 'imap')
         $this->service = $service;
     }
 
+    public function __destruct()
+    {
+        $this->setImapErrors();
+    }
+
+    /**
+     * Returns all of the IMAP errors that have occurred
+     *
+     * @return array
+     */
+    public function getImapErrors()
+    {
+        if(empty($this->imapErrors)) {
+            $this->setImapErrors();
+        }
+
+        return $this->imapErrors;
+    }
+
+    /**
+     * Sets all the IMAP errors that have occurred
+     */
+    public function setImapErrors()
+    {
+        $this->imapErrors = imap_errors();
+    }
+
     /**
```

Comment:
@tedivm The point of the destructor is to clear errors by checking imap_errors() before php goes in shutdown mode. This seems to be a bug in php. Otherwise php would trigger an Unknown on line 0 Notice. See the bug report that I linked